### PR TITLE
Update to ink! v3.0

### PR DESCRIPTION
**Pull request by danforbes**

### CODE REVIEW

File: 1/assets/1.1-finished-code.rs [hunk 1]
```diff
@@ -28,9 +26,11 @@ mod incrementer {
     mod tests {
         use super::*;
 
-        #[test]
+        use ink_lang as ink;
+
+        #[ink::test]
```
Comment (Robbepop) [inline 1/assets/1.1-finished-code.rs:31]:
```suggestion
        #[test]
```

Comment (Robbepop) [reply 1/assets/1.1-finished-code.rs:31]: The special `#[ink::test]` proc macro is only required if the test interacts with the environment, e.g. wants to receive caller, transferred balance etc. This does not count for storage reads/writes though.

File: 1/assets/1.2-finished-code.rs [hunk 2]
```diff
@@ -3,28 +3,30 @@
 
 use ink_lang as ink;
 
-#[ink::contract(version = "0.1.0")]
+#[ink::contract()]
```
Comment (Robbepop) [inline 1/assets/1.2-finished-code.rs:6]:
```suggestion
#[ink::contract]
```

Comment (Robbepop) [reply 1/assets/1.2-finished-code.rs:6]: Please remove the empty parentheses `()` for all code examples. Only use parentheses when they actually enclose parameters.

File: 1/assets/1.2-finished-code.rs [hunk 3]
```diff
@@ -33,9 +35,11 @@ mod incrementer {
     mod tests {
         use super::*;
 
-        #[test]
+        use ink_lang as ink;
+
+        #[ink::test]
```
Comment (Robbepop) [inline 1/assets/1.2-finished-code.rs:40]:
```suggestion
        #[test]
```

Comment (Robbepop) [reply 1/assets/1.2-finished-code.rs:40]: As described above. While I am not sure it is easier to learn when to use `#[test]` and when to use `#[ink::test]` this workshop should teach people the differences.

File: 1/incrementing-my-value.md [hunk 4]
```diff
@@ -16,48 +16,28 @@ impl MyContract {
...
-let caller = self.env().caller();
-match self.my_number_map.get(&caller) {
-    Some(_) => {
-        self.my_number_map.mutate_with(&caller, |value| *value += by);
-    }
-    None => {
-        self.my_number_map.insert(caller, by);
-    }
-};
-```
```
Comment (Robbepop) [inline 1/incrementing-my-value.md:59]: Why was this removed? It is still fundamentally true.

File: 1/incrementing-the-value.md [hunk 5]
```diff
@@ -12,39 +12,17 @@ If you are simply _reading_ from the contract storage, you only need to pass `&s
...
-
-    #[ink(message)]
-    fn my_adder(&mut self, add_value: u32) {
-        self.my_number += add_value;
-    }
-}
-```
-
-However, if the value is not initialized, your contract will compile fine, but will panic during contract execution! **We really cannot understate how easy it is to make mistakes this way.**
-
```
Comment (Robbepop) [inline 1/incrementing-the-value.md:47]: This should still be documented but using the new type `ink_storage::lazy::Lazy<T>` which shares some similarities with the old `ink_core::storage::Value<T>`.

Comment (Robbepop) [reply 1/incrementing-the-value.md:47]: You use `Lazy` to lazily manipulate the underlying value. So if you use a raw `i32` in your storage the contract will load this value upon contract execution directly even if you never use it. A `Lazy<i32>` only loads it if it is actually used. For flipper and incrementer contracts we use `i32` instead of `Lazy<i32>` since all their methods always operate on the `value` so using a `Lazy<i32>` instead wouldn't provide us with any benefit. Also `Lazy<T>` has some minor overhead compared to using `T` directly.

File: 1/storing-a-value.md [hunk 6]
```diff
@@ -32,69 +32,74 @@ Here is an example of how you would store an `AccountId` and `Balance`:
...
+```rust
+pub fn new(init_value: u32) -> Self {
+    Self {
+        number: init_value,
+    }
+}
+```
+
+This can be done anywhere in our contract logic, but most commonly this happens by using the `#[ink(constructor)]` attribute.
+
```
Comment (Robbepop) [inline 1/storing-a-value.md:102]: This tip applied where the signature of an ink! constructor was still `&mut self, ...` but nowadays this rule is guarded by ink! and the Rust compiler and users generally can no longer easily shoot their foot by forgetting to initialize a value since they must initialize everything always since they otherwise run into compile errors.

File: 1/incrementing-my-value.md (hunk 4 above)
Comment (danforbes) [reply 1/incrementing-my-value.md:59]: Can you please suggest a rewrite?

File: 1/storing-a-value.md (hunk 6 above)
Comment (danforbes) [reply 1/storing-a-value.md:102]: Should this section be removed?

Comment (Robbepop) [reply 1/storing-a-value.md:102]: yey

File: 1/incrementing-my-value.md (hunk 4 above)
Comment (Robbepop) [reply 1/incrementing-my-value.md:59]: Nowadays I'd prefer this to be written in terms of the well-known entry API:
https://paritytech.github.io/ink/ink_storage/collections/hashmap/struct.HashMap.html#method.entry

Comment (Robbepop) [reply 1/incrementing-my-value.md:59]: Just a scetch but:
```rust
self.my_number_map.entry(caller).and_mutate(|old_value| old_value += by).or_insert(by);
```

File: 0/setup.md [hunk 7]
```diff
@@ -9,12 +9,17 @@ Follow the
...
+rustup component add rust-src --toolchain nightly
+```
+
 ## Installing A Substrate Node
 
 We need to use a Substrate node with the built-in Contracts module. For this workshop we'll use the pre-designed Substrate node client.
 
 ```bash
-cargo install node-cli --git https://github.com/paritytech/substrate.git --tag v2.0.0 --force --locked
+cargo install canvas-node --git https://github.com/paritytech/canvas-node.git --force
```
Comment (danforbes) [inline 0/setup.md:22]: Are we going to make this repo public?

Comment (riusricardo) [reply 0/setup.md:22]: I was telling this to @Robbepop If not, we end up with this issue https://github.com/paritytech/canvas-ui/issues/6

Comment (Robbepop) [reply 0/setup.md:22]: Yep let us make it public.

File: 1/assets/1.1-finished-code.rs [hunk 8]
```diff
@@ -28,9 +27,11 @@ mod incrementer {
     mod tests {
         use super::*;
 
+        use ink_lang as ink;
```
Comment (ropottnik) [inline 1/assets/1.1-finished-code.rs:30]: this is not really needed at this point to make the tests pass

File: 0/setup.md [hunk 9]
```diff
@@ -35,7 +29,7 @@ The final tool we will be installing is the ink! command line utility which will
 You can install the utility using Cargo with:
 
 ```bash
-cargo install cargo-contract --vers 0.6.2 --force
+cargo install cargo-contract --vers 0.7.0 --force
```
Comment (danforbes) [inline 0/setup.md:32]: When will this be published?

Comment (Robbepop) [reply 0/setup.md:32]: @ascjones will publish it today evening. Maybe as `rc1` though.

File: 0/setup.md (hunk 7 above)
Comment (Robbepop) [reply 0/setup.md:22]: That's an idea but I think it is better to provide a built-in solution to not require this. We could still provide a tip to those who already went through that other tutorial that they can also optionally use their own chain.

Comment (ascjones) [reply 0/setup.md:22]: https://github.com/paritytech/canvas-node/releases/tag/v0.1.0

File: 0/setup.md (hunk 9 above)
Comment (ascjones) [reply 0/setup.md:34]: https://github.com/paritytech/cargo-contract/releases/tag/v0.7.0

Comment (ascjones) [reply 0/setup.md:34]: https://crates.io/crates/cargo-contract/0.7.0

**Comment by shawntabrizi** [comment]
Just jumping in to say nice job! 🚀

**Review by riusricardo — APPROVED** [review]
We are go for launch :rocket:

File: 0/calling-your-contract.md [hunk 10]
```diff
@@ -42,7 +40,7 @@ We will not go over these setup and deployment steps again, but we will use them
 The rest of the tutorial will have **template code** which you will use to walk through the different steps of contract development. Each template comes with a fully designed suite of tests that should pass if you programmed your contract correctly. Before you move on from a section, make sure that you run:
 
 ```bash
-cargo +nightly test
+cargo +nightly-2020-10-08 test
```
Comment (riusricardo) [inline 0/calling-your-contract.md:43]:
```suggestion
cargo +nightly test
```
